cfme/tests/openstack/infrastructure/test_provider.py

```diff
@@ -12,18 +13,27 @@
 ]
 
 
+@pytest.fixture(scope='module')
+def provider_recreate(provider):
+    yield
+    uc = get_crud(provider.name)
+    uc.create()
+
+
 def test_api_port(provider):
     view_details = navigate_to(provider, 'Details')
     port = provider.data['endpoints']['default']['api_port']
     api_port = int(view_details.entities.summary('Properties').get_text_of('API Port'))
     assert api_port == port, 'Invalid API Port'
 
+
 def test_credentials_quads(provider):
     view = navigate_to(provider, 'All')
     prov_item = view.entities.get_entity(name=provider.name, surf_pages=True)
     assert prov_item.data.get('creds') and 'checkmark' in prov_item.data['creds']
 
 
+@pytest.mark.usefixtures('provider_recreate')
```

Comment:
Required: 
@pytest.mark.provider([OpenstackInfraProvider], override=True, selector=ONE)
def test_delete_provider(provider, setup_provider):

this should do provider recreate instead of fixture